linux/tests/sync_linux_ut/sync_linux_ut.c

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdbool.h>
+#endif
+
+#include <sys/syscall.h>
+#include <linux/futex.h>
+#include <time.h>
+#include "azure_macro_utils/macro_utils.h"
+#include "testrunnerswitcher.h"
+
+#include "umock_c/umock_c.h"
+#include "umock_c/umocktypes_windows.h"
+
+#include "sync.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#define ENABLE_MOCKS
+#include "mock_sync.h"
+#undef ENABLE_MOCKS
+
+#ifdef __cplusplus
+}
+#endif
+
+static uint32_t expected_timeout_ms;
+static int expected_return_val;

```

Comment:
>int [](start = 7, length = 3)

looks like a bool...

Reply:
(at least that's what `hook_mock_syscall` returns...)

---
In reply to: [443821362](https://github.com/Azure/azure-c-pal/pull/6#discussion_r443821362) [](ancestors = 443821362)